lib/optimize/ConcatenatedModule.js

```diff
@@ -1271,26 +1258,13 @@ class HarmonyImportSideEffectDependencyConcatenatedTemplate {
 		this.modulesMap = modulesMap;
 	}
 
-	getHarmonyInitOrder(dep) {
+	getInitFragments(dep, source, runtime) {
 		const module = dep._module;
 		const info = this.modulesMap.get(module);
 		if (!info) {
-			return this.originalTemplate.getHarmonyInitOrder(dep);
-		}
-		return NaN;
-	}
-
-	harmonyInit(dep, source, runtime, dependencyTemplates) {
-		const module = dep._module;
-		const info = this.modulesMap.get(module);
-		if (!info) {
-			this.originalTemplate.harmonyInit(
-				dep,
-				source,
-				runtime,
-				dependencyTemplates
-			);
-			return;
+			return this.originalTemplate.getInitFragments(dep, source, runtime);
+		} else {
```

Comment:
Maybe a common base class for all these dependency templates in ConcatenatedModule